release-notes.txt

```diff
@@ -1,3 +1,565 @@
+
+This release includes support for three new boards from the Seeed Studio XIAO
+family, notably the nRF52840 (#20980), nRF52840 Sense (#21332), and the ESP32C3
+variant (#21267). RIOT now also boots on SAM4S Xplained Pro evaluation kits (#21201).
+
+175 pull requests, composed of 357 commits, have been merged since the
+last release, and 12 issues have been solved. 35 people contributed with
+code in 82 days. 1963 files have been touched with 49418 (+) insertions and
+37661 deletions (-).
+
+
+Notations used below
+====================
+
+    + means new feature/item
+    * means modified feature/item
+    - means removed feature/item
+
+
+New features and changes
+========================
+
+Core
+----
+
++ core/cib: Add a prominent warning about thread safety (#21401)
+* core/debug: extend doc about DEBUG() / LOG_DEBUG() (#21276)
+* core/thread: error in thread.c corrected (#21253)
+
+System Libraries
+----------------
+
++ sys/bcd: add bcd_buf_from_u32() (#21308)
++ sys/event: add `event_deferred_callback_post()` helper (#21219)
++ sys/event: add `event_loop_debug` pseudo-module (#20849)
++ sys/fmt: add {fmt,scn}_time_tm_iso8601 (#21391)
++ sys/psa_crypto: Adding hmac hashing on psa_import_key and fix max
+  hmac key size (#21297)
++ sys/ztimer: add hint for setting static sleep adjustment (#21376)
+* drivers/mtd_default: remove deprecated `mtd_default_get_dev()` (#21386)
+* Fido2 update tests (#20876)
+* psa_crypto: fold some long lines (#21303)
+* rust: Simplify Makefiles (#21239)
+* sys/chunked_ringbuffer: let `crb_end_chunk()` return size of the
+  chunk (#20942)
+* sys/isrpipe: unit tests, doc, and fix init (#21344)
+* sys/malloc_tracing: remove deprecated module (#21384)
+* sys/psa_crypto: drop no-op Doxygen statements (#21393)
+* sys/psa_crypto: usa auto_init module for initialization (#21322)
+* sys/vfs: remove deprecated `vfs_iterate_mounts()` (#21387)
+* sys/ztimer: fix small typo in doc (#21208)
+* treewide: update riot-wrappers and riot-sys (#21383)
+
+Networking
+----------
+
++ sys/net/nanocoap: Implement Observe (Server-Side) (#21147)
+* drivers/include: fixing order of doxygen tags and header guards (#21325)
+* ieee802154/submac: calculate symbol time on demand (#19668)
+* lwip: bump to v2.2.1 (#21196)
+* nanocoap: avoid non `static` buffers of configurable size (#21270)
+* nanocoap_sock: implement observe (Client-Side) (#21160)
+* net_sock_async_event: cancel async event on sock_*_close() (#21180)
+* sys/net/gcoap: remove deprecated gcoap_req_send_tl() (#21385)
+* sys/net/ipv6: use markdown links in doc (#21356)
+
+Packages
+--------
+
+* Pkg/ccn-lite: use latest ccn-lite commit (#21279)
+* pkg/driver_cryptocell_310: allow data to be in ROM on hash update (#21214)
+* pkg/driver_cryptocell_310: check for input data to be in RAM (#21138)
+* pkg/esp32_sdk: Fix broken `rtc_io.h` declaration and prevent
+  `sys/uio.h` inclusion outside RIOT (#21408)
+
+Boards
+------
+
++ Add support for Seeed Studio ESP32C3 Xiao (#21267)
```

Comment:
```suggestion
+ boards/seeedstudio-xiao-esp32c3: Add support for Seeed Studio ESP32C3 Xiao (#21267)
```

I was so free to already update the title in https://github.com/RIOT-OS/RIOT/pull/21267 to match this.